services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/SqsBatchFunctions.java

```diff
@@ -0,0 +1,333 @@
+import software.amazon.awssdk.services.sqs.model.DeleteMessageBatchRequest;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageBatchRequestEntry;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageBatchResponse;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageRequest;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageResponse;
+import software.amazon.awssdk.services.sqs.model.SendMessageBatchRequest;
+import software.amazon.awssdk.services.sqs.model.SendMessageBatchRequestEntry;
+import software.amazon.awssdk.services.sqs.model.SendMessageBatchResponse;
+import software.amazon.awssdk.services.sqs.model.SendMessageBatchResultEntry;
+import software.amazon.awssdk.services.sqs.model.SendMessageRequest;
+import software.amazon.awssdk.services.sqs.model.SendMessageResponse;
+import software.amazon.awssdk.utils.BinaryUtils;
+import software.amazon.awssdk.utils.Md5Utils;
+
+@SdkInternalApi
+public final class SqsBatchFunctions {
+
+    private SqsBatchFunctions() {
+    }
+
+    public static BatchAndSend<SendMessageRequest, SendMessageBatchResponse> sendMessageBatchFunction(SqsClient client) {
+        return (identifiedRequests, destination) -> {
+            List<SendMessageBatchRequestEntry> entries = new ArrayList<>(identifiedRequests.size());
+            identifiedRequests.forEach(identifiedRequest -> {
+                String id = identifiedRequest.id();
+                SendMessageRequest request = identifiedRequest.message();
+                entries.add(createSendMessageBatchRequestEntry(id, request));
+            });
+            Optional<AwsRequestOverrideConfiguration> overrideConfiguration = identifiedRequests.get(0)
+                                                                                                .message()
+                                                                                                .overrideConfiguration();
+            SendMessageBatchRequest batchRequest;
+            if (overrideConfiguration.isPresent()) {
+                batchRequest = SendMessageBatchRequest.builder()
+                                                      .queueUrl(destination)
+                                                      .overrideConfiguration(overrideConfiguration.get())
+                                                      .entries(entries)
+                                                      .build();
+            } else {
+                batchRequest = SendMessageBatchRequest.builder()
+                                                      .queueUrl(destination)
+                                                      .entries(entries)
+                                                      .build();
+            }
+            // TODO: Pass client executor in supplyAsync once an executor is added into the client.
+            return CompletableFuture.supplyAsync(() -> client.sendMessageBatch(batchRequest));
```

Comment:
I think we should add executor to the SqsBatchManager builder to allow users to pass it, and in the default sqs batch manager(sqsClient.batchManager()), it will use the existing executor.

Reply:
Added executor into builder.